moodle/Sniffs/Commenting/FileHasSniff.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+// This file is part of Moodle - https://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <https://www.gnu.org/licenses/>.
+
+namespace MoodleHQ\MoodleCS\moodle\Sniffs\Commenting;
+
+use MoodleHQ\MoodleCS\moodle\Util\Tokens;
+use MoodleHQ\MoodleCS\moodle\Util\Docblocks;
+use MoodleHQ\MoodleCS\moodle\Util\TokenUtil;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Files\File;
+use PHPCSUtils\Tokens\Collections;
+use phpDocumentor\Reflection\Types\Collection;
+
+/**
+ * Checks that a file has appropriate tags in either the file, or single artefact block.
+ *
+ * @copyright  2024 Andrew Lyons <andrew@nicols.co.uk>
+ * @license    https://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class FileHasSniff implements Sniff
+{
+    /** @var string The license text expected */
+    protected static $licenseText = 'https://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later';
```

Comment:
Uhm... this can be problematic for plugins, that can use compatible licenses. And also within core, because there are lots of places pointing to "http" and, maybe also, old GPL2 or whatever.

Maybe this should default to empty, so no check is done, and later enable it if core execution is detected... 

Or, alternatively, I find it really similar to [what happened with the TODO/@todo sniff](https://github.com/moodlehq/moodle-cs/blob/main/moodle/Sniffs/Commenting/TodoCommentSniff.php#L36-L75), that we had to make it a configurable (public!) option, plus a config setting to allow 3rd executions to decide what to do. Maybe we can follow that very same approach.

Reply:
I had wondered the same myself but didn't quite come back to it.

I've gone with the same todo-like solution, but defaulting to:
```php
'@https?://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later@'
```

I'm also _not_ auto-fixing any license for now. Just too noisy.
```